implementations/rust/ockam/ockam_core/src/compat.rs

```diff
@@ -210,16 +210,16 @@ pub mod sync {
     pub use alloc::sync::Arc;
     pub use spin::RwLock;
 
-    /// Wrap `spin::Mutex.lock()` as it does not return Option<T> like `std::sync::Mutex`.
+    /// Wrap `spin::Mutex.lock()` as it does not return LockResult<Guard> like `std::sync::Mutex`.
     pub struct Mutex<T>(spin::Mutex<T>);
     impl<T> Mutex<T> {
         /// Creates a new mutex in an unlocked state ready for use.
         pub fn new(value: T) -> Self {
             Mutex(spin::Mutex::new(value))
         }
         /// Acquires a mutex, blocking the current thread until it is able to do so.
-        pub fn lock(&self) -> Option<spin::MutexGuard<'_, T>> {
-            Some(self.0.lock())
+        pub fn lock(&self) -> Result<spin::MutexGuard<'_, T>, ()> {
```

Comment:
How about `core::convert::Infallible` instead of `()`?

Reply:
Updated, tx! 😊